### PR TITLE
DataArray.to_masked_array always returns array

### PR DESCRIPTION
**Pull request by femtotrader**
Should fix https://github.com/xray/xray/issues/684

### CODE REVIEW

File: xray/test/test_dataarray.py [hunk 1]
```diff
@@ -1406,6 +1406,12 @@ def test_to_masked_array(self):
             ma = da.to_masked_array()
             self.assertIsInstance(ma, np.ma.MaskedArray)
 
+        N = 4
```
Comment (shoyer) [inline xray/test/test_dataarray.py]: could you please add a comment noting the github issue this fixes?

**Comment by shoyer** [comment]
It also would be great to have a quick note in "what's new"

**Comment by shoyer** [comment]
Looks great to me, thanks!

**Comment by jhamman** [comment]
cool, thanks for this @femtotrader

**Pull request merged**